x/twasm/contract/incoming_msgs.go

```diff
@@ -196,6 +196,40 @@ func (p *GovProposal) UnmarshalJSON(b []byte) error {
 			}
 			return nil
 		},
+		"migrate_contract": func(b []byte) error {
+			var proxy = struct { // todo: better use wasmvmtypes.MigrateMsg when names match
+				Contract string `json:"contract"`
+				CodeID   uint64 `json:"code_id"`
+				Msg      []byte `json:"msg"`
+			}{}
+			if err := json.Unmarshal(b, &proxy); err != nil {
+				return sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
+			}
+			result.MigrateContract = &wasmtypes.MigrateContractProposal{
+				Contract: proxy.Contract,
+				CodeID:   proxy.CodeID,
+				Msg:      proxy.Msg,
+			}
+			return nil
+		}, "instantiate_contract": func(b []byte) error {
```

Comment:
I'm confused. I wonder how this worked before?

Please check https://github.com/confio/poe-contracts/blob/30972321799e6306a53aaf8ba0d3fb525c74f52f/packages/bindings/src/gov.rs#L42-L55

In particular, there's a `run_as` field. And `msg` is called `init_msg` there.

Are you suggesting that we sync up these names on the `poe_contracts` side?

Reply:
Another minor difference (shouldn't be breaking): `admin` is optional here (wasmvm), but not there (contracts).

Reply:
I don't think this was ever tested or worked before.
The admin (=migrator) is defined as optional in wasmd. It can be empty